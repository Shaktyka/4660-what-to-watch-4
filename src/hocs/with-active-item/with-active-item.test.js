import React from 'react';
import renderer from 'react-test-renderer';

import withActiveItem from './with-active-item.js';

const onefilmData = {
  id: 7,
  title: `Mindhunter`,
  preview: `mindhunter.jpg`,
  genre: `Drama`,
  year: 2018,
  poster: `the-grand-budapest-hotel-poster.jpg`,
  cover: `bg-the-grand-budapest-hotel.jpg`,
  ratingScore: 6.7,
  ratingCount: 410,
  description: [
    `Mindhunter`,
    `The film`,
    `They reach`
  ],
  director: `Bryan Singer`,
  starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  source: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

const mockComponent = () => <div />;

const MockComponentWrapped = withActiveItem(mockComponent);

// Продумать ещё раз набор тестов для хока

describe(`withActiveItem rendering`, () => {

  it(`withActiveItem renders with ...`, ()=>{
    const tree = renderer.create(
        <MockComponentWrapped
          film={onefilmData}
          // дописать
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`withActiveItem renders with ...`, ()=>{
    const tree = renderer.create(
        <MockComponentWrapped
          // дописать
          activeItem={false}
          setActiveItem={()=>{}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
