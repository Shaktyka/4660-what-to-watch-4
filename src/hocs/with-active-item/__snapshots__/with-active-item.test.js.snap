// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveItem rendering withActiveItem renders with ... 1`] = `<div />`;

exports[`withActiveItem rendering withActiveItem renders with ... 2`] = `<div />`;
